core/src/main/java/org/mule/runtime/core/policy/DefaultOperationPolicy.java

```diff
@@ -0,0 +1,93 @@
+ * from the policy to the next operation {@link Event} or from the next operation result to the {@link Event} that must continue
+ * the execution of the policy.
+ * <p>
+ * If a non-empty {@code operationPolicyParametersTransformer} is passed to this class, then it will be used to convert the
+ * message that arrives to the next-operation component of the policy chain to a new {@link Event} containing the data to be used
+ * to execute the provided {@link Processor}.
+ */
+public class DefaultOperationPolicy implements OperationPolicy {
+
+  private final PolicyChain policyChain;
+  private final Optional<OperationPolicyParametersTransformer> operationPolicyParametersTransformer;
+  private final PolicyStateHandler policyStateHandler;
+  private final PolicyEventConverter policyEventConverter = new PolicyEventConverter();
+
+  public DefaultOperationPolicy(PolicyChain policyChain,
+                                Optional<OperationPolicyParametersTransformer> operationPolicyParametersTransformer,
+                                PolicyStateHandler policyStateHandler) {
+    this.policyChain = policyChain;
+    this.operationPolicyParametersTransformer = operationPolicyParametersTransformer;
+    this.policyStateHandler = policyStateHandler;
+  }
+
+  /**
+   * Process the policy chain of processors. The provided {@code nextOperation} function has the behaviour to be executed by the
+   * next-operation of the chain.
+   *
+   * @param operationEvent the event with the data to execute the operation
+   * @param nextProcessor the next-operation processor implementation
+   * @param operationParametersProcessor a processor that converts an {@link Event} to the set of parameters to be sent by the
+   *        operation based on the user configuration.
+   * @return the result of processing the {@code event} through the policy chain.
+   * @throws Exception
+   */
+  @Override
+  public Event process(Event operationEvent, Processor nextProcessor, OperationParametersProcessor operationParametersProcessor)
+      throws Exception {
+    Optional<Event> latestPolicyState = policyStateHandler.getLatestState(operationEvent.getContext().getId());
+    Event variablesProviderEvent = latestPolicyState.orElseGet(() -> Event.builder(operationEvent.getContext()).build());
+    Map<String, Object> originalParametersMap = operationParametersProcessor.getOperationParameters();
+    Message message = operationEvent.getMessage();
+    message = operationPolicyParametersTransformer
+        .map(parametersTransformer -> parametersTransformer.fromParametersToMessage(originalParametersMap)).orElse(message);
+    Event policyEvent = policyEventConverter.createEvent(message, variablesProviderEvent);
+    Processor operationCall = buildOperationExecutionWithPolicyFunction(nextProcessor);
+    policyStateHandler.updateNextOperation(operationEvent.getContext().getId(), operationCall);
```

Comment:
Yes it is. Do you mean because many policies will be store state using the same ID?. If that's the case then you are right. I will fix it.